zebedee-cms/src/test/java/com/github/onsdigital/zebedee/model/approval/task/CollectionPdfGeneratorTest.java

```diff
@@ -40,27 +45,87 @@ public class CollectionPdfGeneratorTest {
+        
+        ContentDetail content = new ContentDetail("Some article", uri, PageType.ARTICLE);
+        content.description.language = null;
+        collectionContent.add(content);
+
+        // Mock there is not a Welsh resource for this content
+        when(contentReader.getResource(uri+"/data_cy.json")).thenThrow(new NotFoundException("Welsh data file not found"));
+
+        generator.generatePDFsForCollection(collection, contentReader, contentWriter, collectionContent);
+
+        // Verify it generates the English version only
+        verify(pdfService, times(1)).generatePdf(contentWriter, uri, ContentLanguage.ENGLISH);
+        verify(pdfService, never()).generatePdf(contentWriter, uri, ContentLanguage.WELSH);
+    }
 
-        ArrayList<ContentDetail> collectionContent = new ArrayList<>();
+    @Test
+    public void shouldGeneratePdfForEnglishOnlyContent() throws IOException, ZebedeeException {
+        List<ContentDetail> collectionContent = new ArrayList<>();
         String uri = "/the/uri";
-        collectionContent.add(new ContentDetail("Some article", uri, PageType.ARTICLE));
+        
+        ContentDetail content = new ContentDetail("Some article", uri, PageType.ARTICLE);
+        content.description.language = ContentLanguage.ENGLISH.getId();
+        collectionContent.add(content);
+
+        // Mock there is not a Welsh resource for this content
+        when(contentReader.getResource(uri+"/data_cy.json")).thenThrow(new NotFoundException("Welsh data file not found"));
+
+        generator.generatePDFsForCollection(collection, contentReader, contentWriter, collectionContent);
+
+        // Verify it generates the English version only
+        verify(pdfService, times(1)).generatePdf(contentWriter, uri, ContentLanguage.ENGLISH);
+        verify(pdfService, never()).generatePdf(contentWriter, uri, ContentLanguage.WELSH);
+    }
+
+    @Test
+    public void shouldGeneratePdfForWelshOnlyContent() throws IOException, ZebedeeException {
+        List<ContentDetail> collectionContent = new ArrayList<>();
+        String uri = "/the/uri";
+        
+        ContentDetail content = new ContentDetail("Some article", uri, PageType.ARTICLE);
+        content.description.language = ContentLanguage.WELSH.getId();
+        collectionContent.add(content);
+
+        generator.generatePDFsForCollection(collection, contentReader, contentWriter, collectionContent);
+
+        // If a Welsh version is requested, it means there is no English version (as it would override it)
+        verify(contentReader, never()).getResource(uri+"/data_cy.json");
+        // Verify it generates the Welsh version only
+        verify(pdfService, times(1)).generatePdf(contentWriter, uri, ContentLanguage.WELSH);
+        verify(pdfService, never()).generatePdf(contentWriter, uri, ContentLanguage.ENGLISH);
+    }
+
+    @Test
+    public void shouldGeneratePdfForEnglishAndWelshContent() throws IOException, ZebedeeException {
```

Comment:
As per other comment, I don't think this is right. If we haven't modified the Welsh version, there's no need to regenerate it. PDF generation is computationally expensive so this could massively reduce approval times and potential issues for especially large collections.